pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -0,0 +1,182 @@
+// and synchronize the Pods and Namespaces affected by Network Policies and enforce
+// their rules.
+
+package networkpolicy
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	networkingv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/controller/networkpolicy/store"
+	antreatypes "github.com/vmware-tanzu/antrea/pkg/controller/types"
+)
+
+type EndpointQuerier interface {
+	QueryNetworkPolicies(namespace string, podName string) (*EndpointQueryResponse, error)
+}
+
+// EndpointQueryReplier is responsible for handling query requests from antctl query
+type EndpointQueryReplier struct {
+	networkPolicyController *NetworkPolicyController
+}
+
+// EndpointQueryResponse is the reply struct for QueryNetworkPolicies
+type EndpointQueryResponse struct {
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+// Endpoint holds response information for an endpoint following a query
+type Endpoint struct {
+	Namespace string   `json:"namespace,omitempty"`
+	Name      string   `json:"name,omitempty"`
+	Policies  []Policy `json:"policies,omitempty"`
+	Rules     []Rule   `json:"rules,omitempty"`
+}
+
+type PolicyRef struct {
+	Namespace string    `json:"namespace,omitempty"`
+	Name      string    `json:"name,omitempty"`
+	UID       types.UID `json:"uid,omitempty"`
+}
+
+// Policy holds network policy information to be relayed to client following query endpoint
+type Policy struct {
+	PolicyRef
+	selector metav1.LabelSelector `json:"selector,omitempty"`
+}
+
+type Rule struct {
+	PolicyRef
+	Direction networkingv1beta1.Direction `json:"direction,omitempty"`
+	RuleIndex int                         `json:"ruleindex,omitempty"`
+}
+
+// NewNetworkPolicyController returns a new *NetworkPolicyController.
+func NewEndpointQueryReplier(networkPolicyController *NetworkPolicyController) *EndpointQueryReplier {
+	n := &EndpointQueryReplier{
+		networkPolicyController: networkPolicyController,
+	}
+	return n
+}
+
+//QueryNetworkPolicies returns kubernetes network policy references relevant to the selected network endpoint. Relevant
+//policies fall into three categories: applied policies (Policies in Endpoint type) are policies which directly apply to
+//an endpoint, egress and ingress rules (Rules in Endpoint type) are policies which reference the endpoint in an ingress/
+//egress rule respectively.
+func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName string) (*EndpointQueryResponse, error) {
+	// check if namespace and podName select an existing pod
+	_, err := eq.networkPolicyController.podInformer.Lister().Pods(namespace).Get(podName)
+	if err != nil {
+		return nil, nil
+	}
+	type ruleTemp struct {
+		policy *antreatypes.NetworkPolicy
+		index  int
+	}
+	// create network policies categories
+	applied, ingress, egress := make([]*antreatypes.NetworkPolicy, 0), make([]*ruleTemp, 0),
+		make([]*ruleTemp, 0)
```

Comment:
this looks not very tidy. we don't need to save lines for this case.
Either 
```
var applied []*antreatypes.NetworkPolicy
var ingress, egress []*ruleTemp
```
or
```
applied := make([]*antreatypes.NetworkPolicy, 0)
ingress := make([]*ruleTemp, 0)
egress := make([]*ruleTemp, 0)
```